test/pkey_meth_kdf_test.c

```diff
@@ -94,10 +107,23 @@ static int test_kdf_hkdf(void)
         TEST_error("EVP_PKEY_CTX_set1_hkdf_key");
         goto err;
     }
-    if (EVP_PKEY_CTX_add1_hkdf_info(pctx, (const unsigned char *)"label", 5)
+    if (index == 0) {
+        if (EVP_PKEY_CTX_add1_hkdf_info(pctx, (const unsigned char *)"label", 5)
             <= 0) {
-        TEST_error("EVP_PKEY_CTX_set1_hkdf_info");
-        goto err;
+            TEST_error("EVP_PKEY_CTX_set1_hkdf_info");
```

Comment:
`EVP_PKEY_CTX_add1_hkdf_info` instead

Reply:
Actually, the wrong functions is called here, and the error messages below this are wrong... but yeah. I need to clean this up.